gobblin-core-base/src/main/java/gobblin/converter/RecordWithMetadataToEncryptedRecordWithMetadataConverter.java

```diff
@@ -19,44 +19,55 @@
 
 package gobblin.converter;
 
-import java.io.UnsupportedEncodingException;
-import java.util.HashMap;
-
-import org.apache.avro.generic.GenericRecord;
-
-import com.google.gson.Gson;
-import com.google.gson.JsonElement;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Map;
 
 import gobblin.configuration.WorkUnitState;
+import gobblin.crypto.EncryptionConfigParser;
+import gobblin.crypto.EncryptionFactory;
 import gobblin.type.RecordWithMetadata;
 import gobblin.type.SerializedRecordWithMetadata;
+import gobblin.writer.StreamCodec;
 
 
 /**
  * A converter that converts a {@link RecordWithMetadata} to a {@link gobblin.type.SerializedRecordWithMetadata}
  * where the serialized bytes represent encrypted data.
  */
-public class EncryptedRecordWithMetadataConverter extends Converter<String, String, RecordWithMetadata, SerializedRecordWithMetadata> {
-  private static final Gson GSON = new Gson();
-
+public class RecordWithMetadataToEncryptedRecordWithMetadataConverter extends Converter<String, String, RecordWithMetadata, SerializedRecordWithMetadata> {
   @Override
-  public String convertSchema(String inputSchema, WorkUnitState workUnit)
-      throws SchemaConversionException {
+  public String convertSchema(String inputSchema, WorkUnitState workUnit) throws SchemaConversionException {
     return "";
   }
 
   @Override
   public Iterable<SerializedRecordWithMetadata> convertRecord(String outputSchema, RecordWithMetadata inputRecord,
-      WorkUnitState workUnit)
-      throws DataConversionException {
+      WorkUnitState workUnit) throws DataConversionException {
 
-    //TODO: Encrypt this
     try {
-      byte[] serialized = GSON.toJson(inputRecord.getRecord()).getBytes("UTF-8");
-      SerializedRecordWithMetadata serializedRecordWithMetadata = new SerializedRecordWithMetadata(serialized,
+      if (!(inputRecord.getRecord() instanceof byte[])) {
+        throw new DataConversionException(
+            "RecordWithMetadataToEncryptedRecordWithMetadataConverter requires input record to be of type byte[]");
+      }
+
+      Map<String, Object> encryptionConfig = EncryptionConfigParser.getConfigForBranch(workUnit);
```

Comment:
Converters support an init method that you can override, where you can do workunit-level instantiation of configuration.